### PR TITLE
Refactor/test connect

### PR DESCRIPTION
**Pull request by myNameIsDu**
Refactor test connect

link: #1737

### CODE REVIEW

**Comment by markerikson** [comment]
Looks okay so far. Lemme know when you have the rest of the logic uncommented and converted!

**Comment by myNameIsDu** [comment]
@markerikson   Because it's a big file, I think it's a good idea for you to check it earlier, and comment out the test cases that are not converted, because they don't pass the check and all of them are temporarily commented out until I convert them little by little, Finally, thank you for giving me valuable writing instructions

**Comment by myNameIsDu** [comment]
> Looks okay so far. Lemme know when you have the rest of the logic uncommented and converted!

ok, thanks

**Comment by myNameIsDu** [comment]
@markerikson I don't know why it uses npm instead and causes an error that triggers yarn, strange

**Comment by markerikson** [comment]
@myNameIsDu looks like commit 8bbc934dcd6648948fd92365c5c21d61563c109d added a `package-lock.json` file.  Delete that and I suspect CSB should be happier.

**Comment by myNameIsDu** [comment]
> @myNameIsDu looks like commit [8bbc934](https://github.com/reduxjs/react-redux/commit/8bbc934dcd6648948fd92365c5c21d61563c109d) added a `package-lock.json` file. Delete that and I suspect CSB should be happier.

Yes, then the default order for the CSB is to check the lock file of npm => yarn  :satisfied: It's an interesting thing, but I don't know much about it

**Comment by myNameIsDu** [comment]
@markerikson  I found some old skip cases, should I delete them?

File: test/components/connect.spec.tsx [hunk 1]
```diff
@@ -2020,6 +2382,7 @@ describe('React', () => {
 
         rtl.render(
           <ProviderMock store={store}>
+            {/*// @ts-ignore */}
```
Comment (markerikson) [inline test/components/connect.spec.tsx:2385]: :question: _just_ out of curiosity, what issues are we ignoring in this area of the code?

File: test/components/connect.spec.tsx [hunk 2]
```diff
@@ -2241,40 +2665,39 @@ describe('React', () => {
...
-              <Wrapper />
-            </ProviderMock>
-          )
-        ).toThrow(
-          `To access the wrapped instance, you need to specify { withRef: true } in the options argument of the connect() call`
-        )
-        spy.mockRestore()
-      })
-
+      //   it.skip('should throw when trying to access the wrapped instance if withRef is not specified', () => {
```
Comment (markerikson) [inline test/components/connect.spec.tsx:2668]: :hand: Yeah, this is dead - `getWrappedInstance()` went away in version 6, and I guess we never removed this test.  Delete it.

**Review by markerikson — CHANGES_REQUESTED** [review]
Looks great! My only comments are a question about what TS errors we're ignoring in a few places, and that the `getWrappedInstance` test can be deleted.

File: test/components/connect.spec.tsx (hunk 2 above)
Comment (myNameIsDu) [reply test/components/connect.spec.tsx:2668]: image
This descript is also all skip, I can also delete it directly?

Comment (markerikson) [reply test/components/connect.spec.tsx:2668]: yeah, let's delete that one too

File: test/components/connect.spec.tsx (hunk 1 above)
Comment (markerikson) [reply test/components/connect.spec.tsx:2385]: Hmm. Yeah, it's entirely possible that's a bug in the types.

Let's leave the ignore in there for now, and maybe leave a `// TODO` comment saying to re-investigate it later?

Comment (myNameIsDu) [reply test/components/connect.spec.tsx:2385]: That's a good idea. I think I can fix it after this PR #1771  is done

**Comment by markerikson** [comment]
Awright, let's get this in.

Thank you so much for working on this!

**Pull request merged**